### DISCUSSION TITLE
[@types/jsdom] ConstructorOptions -> BaseOptions -> contentType is too restrictive

### DISCUSSION THREAD

**Discussion by CodingDoug** [DefinitelyTyped/DefinitelyTyped#69511] [Issues with a @types package]
The constructor options property is declared like this:

```
contentType?: SupportedContentTypes | undefined;

    type SupportedContentTypes =
        | "text/html"
        | "application/xhtml+xml"
        | "application/xml"
        | "text/xml"
        | "image/svg+xml";
```

This means that contentType must only be one of the acceptable strings.  However, this is too restrictive.  The contentType attribute can also take a charset along with it (e.g. `text/html; charset=utf-8`).  JSDOM observes this charset correctly, and is even [documented](https://github.com/jsdom/jsdom?tab=readme-ov-file#simple-options) as such (emphasis mine):

> contentType affects the value read from document.contentType, as well as how the document is parsed: as HTML or as XML. Values that are not a [HTML MIME type](https://mimesniff.spec.whatwg.org/#html-mime-type) or an [XML MIME type](https://mimesniff.spec.whatwg.org/#xml-mime-type) will throw. It defaults to "text/html". **If a charset parameter is present, it can affect [binary data processing](https://github.com/jsdom/jsdom?tab=readme-ov-file#encoding-sniffing).**

My use cases requires that I provide a charset.  Right now, I have to play games with TypeScript to get it to accept my string which is unknown at compile time (since it comes from an HTTP header).

I think `contentType` should accept any string, and the caller should accept that JSDOM will throw (as documented above) if it's not supported.

[a comment by an automated account is not shown]

**Comment by ghost** [comment]
Created #69520 to fix this
